File: AkkaRouterExample/src/main/java/org/akka/essentials/router/example2/custom/Example3.java
package org.akka.essentials.router.example2.custom;

import org.akka.essentials.router.example.MsgEchoActor;

import akka.actor.ActorRef;
import akka.actor.ActorSystem;
import akka.actor.Props;
import akka.routing.DefaultResizer;
import akka.routing.RandomRouter;

public class Example3 {

	/**
	 * @param args
	 * @throws InterruptedException
	 */
	public static void main(String[] args) throws InterruptedException {
		ActorSystem _system = ActorSystem.create("CustomRouteeRouterExample");

		int lowerBound = 2;
		int upperBound = 15;
		DefaultResizer resizer = new DefaultResizer(lowerBound, upperBound);

		ActorRef randomRouter = _system.actorOf(new Props(MsgEchoActor.class)
				.withRouter(new RandomRouter(resizer)));

		for (int i = 1; i <= 10; i++) {
			// sends randomly to actors
			randomRouter.tell(i);
		}
		_system.shutdown();
	}

}

File: AkkaRouterExample/src/main/scala/org/akka/essentials/router/example2/custom/Example3.scala
package org.akka.essentials.router.example2.custom
import akka.routing.DefaultResizer
import akka.actor.ActorSystem
import akka.actor.Props
import org.akka.essentials.router.example.MsgEchoActor
import akka.routing.RandomRouter

object Example3 {
  def main(args: Array[String]): Unit = {
    val _system = ActorSystem.create("CustomRouteeRouterExample")

    val resizer = DefaultResizer(lowerBound = 2, upperBound = 15)

    val randomRouter = _system.actorOf(Props[MsgEchoActor].withRouter(RandomRouter(resizer = Some(resizer))))
    1 to 10 foreach {
      i => randomRouter ! i
    }
    _system.shutdown()
  }
}	